libs/wingc/src/type_check.rs

```diff
@@ -2522,6 +2531,8 @@ impl<'a> TypeChecker<'a> {
 			for arg_expr in variadic_arg_list.iter() {
 				self.validate_type(variadic_array_inner_type, variadic_array_inner_type, *arg_expr);
 			}
+
+			// ^ what does this do when actual type and expected type are always the same variadic array inner type?
```

Comment:
Looks like a bug, feels like we should have used here the type of `arg_expr` as the actual type.

Reply:
I think I managed to fix it in:
https://github.com/winglang/wing/pull/4300